cep-XXXX.md

```diff
@@ -0,0 +1,291 @@
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CEP standardizes the format of `environment.yml` input files for conda clients.
+
+## Motivation
+
+The motivation of this CEP is merely informative. It describes the details of an existing file
+format.
+
+## Nomenclature
+
+The `environment.yml` file format was introduced by `conda env`. It has been traditionally referred
+to as "the YAML input file", but other YAML files exist in the ecosystem now. To avoid confusion,
+in this CEP we will only use the term `environment.yml`. Do note that the actual filename of an
+input file using this format might be a different one (e.g. `env.yml`).
+
+## Specification
+
+`environment.yml` files are YAML documents that encode the necessary information to create a new
+conda environment. The core standard recognizes five top-level keys:
+
+- `name`: The preferred name for the environment.
+- `prefix`: The preferred full path to the environment. Often ignored.
+- `dependencies`: List of package specifications that MUST be installed in the new environment.
+- `channels`: conda channels that will be used to resolve the dependencies.
+- `variables`: Environment variables that SHOULD be added to the `conda-meta/state` file in the
+  resulting environment.
+
+The file name is not standardized, but the extension MUST be `yml` or `yaml`.
+
+Notably, this version of the specification does not feature a `version` (or equivalent) top-level
+key. Future revisions MAY include such a field. As such, the version information of this revision
+is omitted and considered to be `1`.
+
+### Optional extensions
+
+A number of extensions have been implemented by some conda clients. These are not guaranteed to be
+available in all conda clients.
+
+Two additional top-level keys MAY be present:
+
+- `platforms`: Target subdirs for the solver.
+- `category`: Free-form field to label the dependencies as a group.
+
+Preprocessing selectors MAY be available to specify line filters to allow platform-specific values,
+among other uses. There are two types: comment-based and dictionary-based. Only one type of filter
+SHOULD be used in the same document.
+
+### Other top-level keys
+
+Additional top-level keys can be present, but they MUST be ignored and the user SHOULD receive an
+informative warning about them when the conda client does not support them.
+
+### Schema
+
+#### `name` and `prefix`
+
+Optional, `str`.
+
+Both fields refer to the _preferred_ name or path for the newly created environment. Tools SHOULD
+allow these suggestions to be overridden by the user with additional CLI flags or equivalent. If
+the proposed environment path exists, tools MUST NOT overwrite silently by default.
+
+Special names `base` and `root` SHOULD NOT be accepted. Prefixes targeting protected system paths
+SHOULD be rejected. Paths can contain tildes (`~`) and environment variables, and they MUST be
+expanded when present.
```

Comment:
> Paths can contain tildes (`~`) and environment variables, and they MUST be
expanded when present.

Should we clarify this statement and say `~` MUST BE expanded only when it occurs as the first character? Most *nix shells won't expand, e.g., `/tmp/~/subdir-a`.